dcpy/test/lifecycle/builds/test_load.py

```diff
@@ -83,15 +106,174 @@ def test_import_parquet(self, _patched):
             df_inserted_actual,
             table_insert_name_actual,
         ) = pg_mock.insert_dataframe.call_args_list[0][0]
-        assert df_inserted_actual.equals(_mock_preprocessor(ds.name, _test_df))
+        assert df_inserted_actual.equals(_mock_preprocessor(ds.id, _test_df))
         assert table_insert_name_actual == ds.import_as
 
+    @patch("dcpy.connectors.edm.recipes.read_df", return_value=_test_df)
+    def test_import_parquet_df(self, read_df):
+        ds = InputDataset(
+            id="test",
+            version="1",
+            import_as="new_table_name",
+            file_type=recipes.DatasetType.parquet,
+            destination=InputDatasetDestination.df,
+        )
+        res = load.import_dataset(ds, None)
+        assert isinstance(res.destination, pd.DataFrame)
+        assert _test_df.equals(res.destination)
+
+    def test_import_parquet_file(self):
+        recipes.fetch_dataset = MagicMock(side_effect=_mock_fetch_parquet)
+
+        ds = InputDataset(
+            id="test",
+            version="1",
+            import_as="new_table_name",
+            file_type=recipes.DatasetType.parquet,
+            destination=InputDatasetDestination.file,
+        )
+        res = load.import_dataset(ds, None)
+        assert isinstance(res.destination, Path)
+        assert res.destination.exists()
+
     def test_no_version_fails(self):
-        ds = InputDataset(name="test")
-        with pytest.raises(Exception, match="Cannot import"):
-            load.import_dataset(ds, MagicMock)
+        ds = InputDataset(id="test")
+        with pytest.raises(
+            Exception, match="Cannot import a dataset without a resolved version"
+        ):
+            load.import_dataset(ds, None)
 
     def test_unresolved_version_fails(self):
-        ds = InputDataset(name="test", version="latest")
-        with pytest.raises(Exception, match="Cannot import"):
-            load.import_dataset(ds, MagicMock)
+        ds = InputDataset(id="test", version="latest")
+        with pytest.raises(
+            Exception, match="Cannot import a dataset without a resolved version"
+        ):
+            load.import_dataset(ds, None)
+
+    def test_unresolved_filetype(self):
+        ds = InputDataset(id="test", version="1")
+        with pytest.raises(
+            Exception, match="Cannot import a dataset without a resolved file type"
+        ):
+            load.import_dataset(ds, None)
+
+
+@patch("dcpy.connectors.edm.recipes.fetch_dataset", return_value=RESOURCES_DIR)
```

Comment:
I don't think I've ever seen so many patches 😳

Maybe time to think about integration tests, vs unit test?

Reply:
Mayyyybe